hr_attendance_hour_type_report/models/hr_attendance.py

```diff
@@ -0,0 +1,177 @@
+        string="Night hours", compute="_compute_worked_hours", store=True
+    )
+    worked_hours_daytime = fields.Float(
+        string="Day hours",
+        compute="_compute_worked_hours",
+        store=True,
+        readonly=True,
+    )
+
+    worked_hours_overtime = fields.Float(
+        string="Overtime hours",
+        compute="_compute_worked_hours",
+        store=True,
+        readonly=True,
+    )
+
+    date = fields.Date(
+        help="date of the attendance, from the payroll point of view",
+        compute="_compute_date",
+        store=True,
+    )
+    date_type = fields.Selection(
+        [("normal", "Weekday"), ("sunday", "Sunday"), ("holiday", "Public Holiday")],
+        compute="_compute_date_type",
+        store=True,
+    )
+
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        default=lambda self: self.env.company,
+    )
+
+    hr_attendance_overtime = fields.Boolean(related="company_id.hr_attendance_overtime")
+
+    allow_weighting_nighttime_hours = fields.Boolean(
+        related="company_id.allow_weighting_nighttime_hours"
+    )
+
+    allow_weighting_overtime_hours = fields.Boolean(
+        related="company_id.allow_weighting_overtime_hours"
+    )
+
+    weighting_worked_nighttime_hours = fields.Float(
+        compute="_compute_weighting_hours",
+        store=True,
+    )
+    weighting_worked_overtime_hours = fields.Float(
+        compute="_compute_weighting_hours",
+        store=True,
+    )
+
+    @api.depends(
+        "company_id.weighting_nighttime_hours", "company_id.weighting_overtime_hours"
+    )
+    def _compute_weighting_hours(self):
+        for rec in self:
+            nighttime_hours = rec.worked_hours_nighttime
+            overtime_hours = rec.worked_hours_overtime
+            if rec.allow_weighting_nighttime_hours:
+                nighttime_hours *= rec.company_id.weighting_nighttime_hours
+            if rec.allow_weighting_overtime_hours:
+                overtime_hours *= rec.company_id.weighting_overtime_hours
+            rec.weighting_worked_nighttime_hours = nighttime_hours
+            rec.weighting_worked_overtime_hours = overtime_hours
+
+    @api.depends("date")
+    def _compute_date_type(self):
+        for rec in self:
+            if rec.date.weekday() == 6:
+                rec.date_type = "sunday"
+            elif self.env["hr.holidays.public"].is_public_holiday(
+                rec.date, rec.employee_id.id
+            ):
+                rec.date_type = "holiday"
+            else:
+                rec.date_type = "normal"
+
+    @api.depends("check_in")
+    def _compute_date(self):
+        for rec in self:
+            TZ = pytz.timezone(rec.employee_id.tz)
+            tz = rec.employee_id.tz
+            check_in = rec.check_in
+            if not check_in.tzinfo:
+                check_in = TZ.localize(check_in)
+            check_in_tz = check_in.astimezone(pytz.timezone(tz))
+            rec.date = check_in_tz.date()
```

Comment:
```suggestion
            tz_code = rec.employee_id.tz
            TZ = pytz.timezone(tz_code)            
            check_in = rec.check_in
            if not check_in.tzinfo:
                check_in = TZ.localize(check_in)
            check_in_tz = check_in.astimezone(TZ)
            rec.date = check_in_tz.date()
```
Just some refactoring.